### PR TITLE
[SYCL] Set concrete values for enums

### PR DESCRIPTION
**Pull request by alexbatashev**
Changing the order of enum values breaks ABI while preserving API. This
patch adds concrete values to enum items, so that it is possible to
re-order them, or add something in between.

### CODE REVIEW

**Review by romanovvlad — APPROVED** [review]

**Review by turinevgeny — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Comment by alexbatashev** [comment]
@jbrodman friendly ping

File: sycl/include/CL/sycl/ONEAPI/atomic_enums.hpp [hunk 1]
```diff
@@ -23,12 +23,13 @@ namespace sycl {
 namespace ONEAPI {
 
 enum class memory_order : int {
-  relaxed,
-  acquire,
-  __consume_unsupported, // helps optimizer when mapping to std::memory_order
-  release,
-  acq_rel,
-  seq_cst
+  relaxed = 0,
```
Comment (jbrodman) [inline sycl/include/CL/sycl/ONEAPI/atomic_enums.hpp:26]: Do these match their std:: versions?

**Review by jbrodman — APPROVED** [review]
LGTM barring my one question.

File: sycl/include/CL/sycl/ONEAPI/atomic_enums.hpp (hunk 1 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/ONEAPI/atomic_enums.hpp:26]: According to https://en.cppreference.com/w/cpp/atomic/memory_order it does not. However, we can't do anything at the moment. Changing order of these enums would be a breaking change.

**Pull request merged**